### PR TITLE
Print valid container names when the command not specify the container

### LINKED ISSUE #12863: Pod log error message "must select container name" should print valid container names
```
$ kubectl logs master-1-e6bup
error: POD master-1-e6bup has more than one container; please specify the container to print logs for
```

Should try harder to be useful to the user:

```
error: Pod master-1-e6bup has the following containers: api, controllers; please specify the container to print logs for with -c`
```

Also, `POD` capitalized is bad

### PR DESCRIPTION
**Pull request by feihujiang**
Fixes #12863

### CODE REVIEW

**Comment by feihujiang** [comment]
@smarterclayton  Could you please review this PR?

**Comment by smarterclayton** [comment]
Example output?  What is %q displaying for multiple names?

**Comment by feihujiang** [comment]
The example output is 
error: Pod foo has the following containers: ["bar" "foo"]; please specify the container to print logs for with -c

**Comment by feihujiang** [comment]
My test: 
$ ./cluster/kubectl.sh logs kubectl-tester 
error: Pod kubectl-tester has the following containers: ["bb" "kubectl"]; please specify the container to print logs for with -c

**Comment by smarterclayton** [comment]
Hrm - I'd prefer strings.Join(containers, ", ")

On Sep 1, 2015, at 4:16 AM, feihujiang <notifications@github.com> wrote:

My test:
$ ./cluster/kubectl.sh logs kubectl-tester
error: Pod kubectl-tester has the following containers: ["bb" "kubectl"];
please specify the container to print logs for with -c

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/13366#issuecomment-136631383>
.

**Comment by feihujiang** [comment]
@smarterclayton OK, and now The example output is:
Pod foo has the following containers: bar, foo; please specify the container to print logs for with -c

File: pkg/kubectl/cmd/log.go [hunk 1]
```diff
@@ -116,15 +117,18 @@ func RunLog(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
...
 	}
 
-	var container string
-	if cmdutil.GetFlagString(cmd, "container") != "" {
-		// [-c CONTAINER]
-		container = p.containerName
-	} else {
+	// [-c CONTAINER]
+	container := p.containerName
+	if container == "" {
```
Comment (smarterclayton) [inline pkg/kubectl/cmd/log.go]: `len(container) == 0` (convention)

**Comment by smarterclayton** [comment]
Changes LGTM after you fix the nit.

**Comment by feihujiang** [comment]
Fixed the nit

**Pull request merged**